### ISSUE TITLE
Modify `evaluation_data` to `validation_data` in args.

### ISSUE THREAD

**Issue by workingloong** [sql-machine-learning/elasticdl#1355]
we should modify `evaluation_data` to `validation_data` in args, because evaluation_data` is confusion.

The explanations of "validation set" in Wiki are:
1. "The validation dataset provides an unbiased **evaluation** of a model fit on the training dataset while tuning the model's hyperparameters (e.g. the number of hidden units in a neural network). Validation datasets can be used for regularization by early stopping: stop training when the error on the validation dataset increases, as this is a sign of overfitting to the training dataset."
2. "The terms test set and validation set are sometimes used in a way that flips their meaning in both industry and academia. In the erroneous usage, "test set" becomes the development set, and "validation set" is the independent set used to **evaluate** the performance of a fully specified classifier."

The explanation of "test dataset" in [Wiki] is:
1. "The test dataset is a dataset used to provide an unbiased **evaluation** of a final model fit on the training dataset."
